client/app/components/Notifier.jsx

```diff
@@ -0,0 +1,106 @@
+/** @jsx h */
+import { h, Component } from 'preact'
+
+const createStore = () => {
+  let notifications = []
+  let listeners = []
+
+  const dispatch = (notification) => {
+    notification.id = notifications.length
+    notifications.push(notification)
+    listeners.forEach(listener => listener(notification))
+  }
+
+  const subscribe = (listener) => {
+    listeners.push(listener)
+  }
+
+  return { dispatch, subscribe }
+}
+
+const store = createStore()
+
+class Notification extends Component {
+  componentDidMount () {
+    this.closeTimer = setTimeout(() => {
+      this.beginClosing()
+    }, 2000)
+  }
+
+  componentWillUnmount () {
+    if (this.closeTimer) {
+      clearTimeout(this.closeTimer)
+    }
+  }
+
+  beginClosing () {
+    const onTransitionEnd = () => this.close()
+    this.base.addEventListener('transitionend', onTransitionEnd, false)
+    this.base.classList.add('coz-notification--closing')
+  }
+
+  close () {
+    this.props.onClose()
+  }
+
+  render () {
+    const { message, type } = this.props
+    return (
+      <div className={['coz-notification', 'coz-notification--' + type].join(' ')}>
+        <div className='coz-notification-title'>{message}</div>
+      </div>
+    )
+  }
+}
+
+export default class Notifier extends Component {
+  constructor (props, context) {
+    super(props, context)
+    this.state = {
+      notifications: []
+    }
+  }
+
+  static info (msg) {
+    store.dispatch({ type: 'info', msg })
+  }
+
+  static warning (msg) {
+    store.dispatch({ type: 'warning', msg })
+  }
+
+  static error (msg) {
+    store.dispatch({ type: 'error', msg })
+  }
+
+  componentDidMount () {
+    store.subscribe(this.notify.bind(this))
+  }
+
+  notify (notification) {
+    this.setState({
+      notifications: [...this.state.notifications, notification]
+    })
+  }
+
+  handleClose (id) {
+    let idx
+    this.state.notifications.forEach((n, i) => { if (n.id === id) idx = i }) // le find du pauvre...
```

Comment:
I think it's better to use polyfills if/when needed. We could use this function anywhere else in the app. After discussions with @gregorylegarec, polyfills can be handled by a simple webpack plugin, so that could be our simple compatibilities solution here.